test/api/requests/site.test.js

```diff
@@ -545,5 +545,40 @@ describe("Site API", () => {
         done()
       }).catch(done)
     })
+
+    it("should update attributes when the value in the request body is an empty string", done => {
+      let site
+      const userPromise = factory.user()
+      const sitePromise = factory.site({
+        users: Promise.all([userPromise]),
+        config: "old-config: true",
+        domain: "https://example.com",
+      })
+      const cookiePromise = session(userPromise)
+
+      Promise.props({
+        user: userPromise,
+        site: sitePromise,
+        cookie: cookiePromise,
+      }).then(results => {
+        site = results.site
+
+        return request("http://localhost:1337")
+          .put(`/v0/site/${site.id}`)
+          .send({
+            config: "",
+            domain: "",
+          })
+          .set("Cookie", results.cookie)
+          .expect(200)
+      }).then(response => {
+        validateAgainstJSONSchema("PUT", "/site/{id}", 200, response.body)
+        return Site.findById(site.id)
+      }).then(site => {
+        expect(site.config).to.equal("")
+        expect(site.domain).to.equal("")
```

Comment:
Would be nice to have a line here to validate we're keeping the site values if not given in the request e.g.
`expect(site.defaultBranch).to.equal("foo")`
where `foo` is set in our site setup code above.